File: MEDIUM/647. Palindromic Substrings/solution.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        count = 0
        for i in range(n):
            l = i
            r = i
            if l>=0 and r<n:
                while l>=0 and r<n and s[l]==s[r]:
                    l-=1
                    r+=1
                    count+=1
        for i in range(n):
            l = i
            r = i+1
            if l>=0 and r<n:
                while l>=0 and r<n and s[l]==s[r]:
                    l-=1
                    r+=1
                    count+=1
        return count 
        
        

File: MEDIUM/647. Palindromic Substrings/solution.cpp
class Solution {
public:
    int countSubstrings(string s) {
       int n = s.size();
       int count = 0;

       for(int i = 0; i<n; i++){
        int l = i; int r = i;
        if(l>=0 and r<n){
        while(l>=0 and r<n and s[l]==s[r]) {
                l--; r++;
                count++;
            }
          }
        }
       
    
        for(int i = 0; i<n; i++){
            int l = i; int r = i+1;
            if(l>=0 and r<n){
                while(l>=0 and r<n and s[l] == s[r]){
                    l--; r++;
                    count++;
                }
            }
        }
       
       return count;
       
    }
};
